DEV/org.openl.core/src/org/openl/binding/impl/MethodNodeBinder.java

```diff
@@ -88,78 +94,141 @@ public IBoundNode bind(ISyntaxNode node, IBindingContext bindingContext) throws
+            IBoundNode iBoundNode = makeDatatypeConstructor(node, bindingContext, type);
+            if (iBoundNode != null) {
+                return iBoundNode;
             }
+        }
 
-            throw new MethodNotFoundException(methodName, parameterTypes);
+        if (!syntaxNodeExceptions.isEmpty()) {
+            syntaxNodeExceptions.forEach(bindingContext::addError);
+            return new ErrorBoundNode(node);
         }
 
-        // There are no other variants - so error.
         throw new MethodNotFoundException(methodName, parameterTypes);
     }
 
+    private IBoundNode makeDatatypeConstructor(ISyntaxNode node, IBindingContext bindingContext, IOpenClass type) {
+        List<IBoundNode> params = new ArrayList<>();
+        List<String> namedParams = new ArrayList<>();
+        boolean isAllParamsAssign = true;
+        try {
+            bindingContext.pushLocalVarContext();
+            ILocalVar localVar = bindingContext
+                .addVar(ISyntaxConstants.THIS_NAMESPACE, bindingContext.getTemporaryVarName(), type);
+            TypeBindingContext varBindingContext = TypeBindingContext.create(bindingContext, localVar, 1);
+            for (int i = 0; i < node.getNumberOfChildren() - 1; i++) {
+                ISyntaxNode child = node.getChild(i);
+                String childType = child.getType();
+                if ("op.assign".equals(childType)) {
+                    IBoundNode iBoundNode = bindChildNode(child, varBindingContext);
+                    String paramName = child.getChild(0).getText();
+                    if (namedParams.contains(paramName)) {
+                        isAllParamsAssign = false;
+                    }
+                    namedParams.add(paramName);
+                    params.add(iBoundNode);
+                } else {
+                    isAllParamsAssign = false;
+                    IBoundNode iBoundNode = bindChildNode(child, bindingContext);
+                    params.add(iBoundNode);
+                }
+            }
+            if (isAllParamsAssign && namedParams.stream().allMatch(n -> type.getField(n) != null)) {
+                IMethodCaller methodCaller = MethodSearch.findConstructor(IOpenClass.EMPTY, bindingContext, type);
+                MethodBoundNode methodBoundNode = new MethodBoundNode(node, methodCaller);
+                return new ConstructorNamedParamsNode(localVar, methodBoundNode, params.toArray(IBoundNode.EMPTY));
+            } else if (!Date.class.getName().equals(type.getName())) {
+                IBoundNode[] children = params.toArray(IBoundNode.EMPTY);
+                if (hasErrorBoundNode(children)) {
+                    return new ErrorBoundNode(node);
+                }
+                IOpenClass[] types = getTypes(children);
+                IMethodCaller methodCaller = MethodSearch.findConstructor(types, bindingContext, type);
+                BindHelper.checkOnDeprecation(node, bindingContext, methodCaller);

```

Comment:
I think this call must be moved inside ``if`` statement below. Useless call if ```methodCaller``` is null